### ISSUE TITLE
Investigate dropping OvUI to follow the IMGUI paradigm

### ISSUE THREAD

**Issue by adriengivry** [adriengivry/Overload#293]
**Problem this feature should fix**
While `OvUI` works fairly well, it adds an abstraction layer that it often not necessary, and require to write more complex code than if the underlying ImGUI library was used directly.

**Expected solution**
We should look into the IMGUI paradigm, and evaluate whether we should move away from `OvUI` to using `ImGUI` directly in `OvEditor`.

**Comment by maxbrundev** [comment]
Definitely, in my opinion the encapsulation is great especially widgets side but the architecture is adding an unnecessary layer of complexity and it's too much event based to update the widgets data.

**Comment by adriengivry** [comment]
Let's close this one. I think we're set on staying with a non-immediate mode approach.
